bob-access/src/authenticator/basic.rs

```diff
@@ -0,0 +1,72 @@
+use std::{collections::HashMap, net::IpAddr};
+
+use crate::{credentials::Credentials, error::Error};
+
+use super::{users_storage::UsersStorage, Authenticator};
+
+#[derive(Debug, Default, Clone)]
+pub struct Basic<Storage: UsersStorage> {
+    users_storage: Storage,
+    nodes: HashMap<IpAddr, Credentials>,
```

Comment:
Node address can be specified not only as an Ip-address, but also as a hostname. Does the current logic handle hostnames properly?

Reply:
Should work now